File: memory-domain/src/main/java/com/memory/domain/routine/repository/RoutineRepositoryCustom.java
package com.memory.domain.routine.repository;

import com.memory.domain.member.Member;
import com.memory.domain.routine.Routine;

import java.util.List;
import java.util.Optional;

public interface RoutineRepositoryCustom {
    
    List<Routine> findActiveRoutinesByMember(Member member);
    
    List<Routine> findAllRoutinesByMember(Member member);
    
    Optional<Routine> findByIdAndMember(Long id, Member member);
}

File: memory-domain/src/main/kotlin/com/memory/domain/routine/repository/RoutineRepositoryCustom.kt
package com.memory.domain.routine.repository

import com.memory.domain.member.Member
import com.memory.domain.routine.Routine
import java.util.Optional

interface RoutineRepositoryCustom {

    fun findActiveRoutinesByMember(member: Member?): List<Routine>

    fun findAllRoutinesByMember(member: Member?): List<Routine>

    fun findByIdAndMember(id: Long?, member: Member?): Optional<Routine>
}
